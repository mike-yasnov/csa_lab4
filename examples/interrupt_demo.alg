// Демонстрация системы прерываний (упрощённая версия)

var interrupt_count = 0;

function main() {
    print("Демонстрация системы прерываний");
    
    var counter = 0;
    
    // Основной цикл работы
    while (counter < 100) {
        counter = counter + 1;
        
        // Периодически выводим состояние
        if (counter % 20 == 0) {
            print("Main loop: counter=");
            print(counter);
        }
        
        // Имитация полезной работы
        var dummy = counter * 2 + 1;
    }
    
    print("Основной цикл завершён");
}

main(); 