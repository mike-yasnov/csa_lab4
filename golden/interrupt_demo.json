{
  "return_code": 0,
  "stdout": "Загрузка программы: /tmp/tempfile/program.bin\nЗагружено 50 инструкций\nЗагрузка данных: /tmp/tempfile/program_data.bin\nЗагружено 136 байт данных\nЗапуск выполнения (максимум 1000000 тактов)...\n\n=== РЕЗУЛЬТАТЫ ВЫПОЛНЕНИЯ ===\nСостояние: halted\nВыполнено инструкций: 6\nЗатрачено тактов: 16\nФинальный PC: 5\n\nВЫВОД ПРОГРАММЫ:\n[208][148][208][181][208][188][208][190][208][189][209][129][209][130][209][128][208][176][209][134][208][184][209][143] [209][129][208][184][209][129][209][130][208][181][208][188][209][139] [208][191][209][128][208][181][209][128][209][139][208][178][208][176][208][189][208][184][208][185]\n\nПрограмма завершена успешно.\n",
  "stderr": ""
}